### PR TITLE
Cudos 946: Improve delpoy functions user experience

### PR DESCRIPTION
**Pull request by aemil145**
Changes:
- Optimised deploy functions parameters
- Removed the option to get a local contract and assign a default signer
- Fixed a bug where accounts.json is absent and not needed but an error is thrown "accounts.json not found"

### CODE REVIEW

**Review by Holymir — APPROVED** [review]

File: packages/blast-core/lib/bre.js [hunk 1]
```diff
@@ -30,41 +25,27 @@ globalThis.bre.getSigners = async function() {
 globalThis.bre.getCustomSigners = async function(privateAccountName) {
   const privateAccounts = getPrivateAccounts()
   if (privateAccountName) {
-    return await getSigner(privateAccounts[privateAccountName])
+    return await getSigner(nodeUrl, privateAccounts[privateAccountName].mnemonic)
```
Comment (maptuhec) [inline packages/blast-core/lib/bre.js]: await is redundant in return statement

File: packages/blast-core/lib/cudos-contract.js [hunk 2]
```diff
@@ -19,18 +23,27 @@ module.exports.CudosContract = class CudosContract {
...
     this.#signer = signer
     this.#contractAddress = deployedContractAddress
-    this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractLabel}.wasm`)
     this.#gasPrice = GasPrice.fromString(getGasPrice())
 
-    if (deployedContractAddress === null && !fs.existsSync(this.#wasmPath)) {
-      throw new BlastError(`Contract with label ${contractLabel} was not found, did you compile it?`)
+    if (deployedContractAddress === null) {
+      this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractLabel}.wasm`)
+      if (!fs.existsSync(this.#wasmPath)) {
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js:30]: why did you change the structure?

File: packages/blast-core/lib/cudos-contract.js [hunk 3]
```diff
@@ -19,18 +23,27 @@ module.exports.CudosContract = class CudosContract {
...
+      }
     }
   }
 
-  async deploy(initMsg, signer = this.#signer, label = this.#contractLabel, funds) {
-    this.#signer = signer
+  async deploy(initMsg, options = {}) {
+    if (options.signer) {
+      this.#signer = options.signer
+    } else if (this.#contractAddress === null) {
```
Comment (maptuhec) [inline packages/blast-core/lib/cudos-contract.js]: please try to remove the else if structure to become more clear, readable, and easy to maintan

**Review by maptuhec — CHANGES_REQUESTED** [review]

File: packages/blast-core/lib/cudos-contract.js (hunk 2 above)
Comment (aemil145) [reply packages/blast-core/lib/cudos-contract.js:30]: changed it to avoid getting wasm path in case `getContractFromAddress()` is used to create contract object. In such a case we don't need a local path to a contract with the same name as the already deployed one

**Review by maptuhec — APPROVED** [review]

**Review by tgntr — APPROVED** [review]
